### ISSUE TITLE
Duplicati 2.0.0.81 - Casting error after first backup

### ISSUE THREAD

**Issue by C0ReDuMP79** [duplicati/duplicati#1296]
Hello,
yesterday we made the first backup with the latest version of Duplicati, the file scan took a lot of time to complete but seems to be normal because the database format changed. The backup ended successfully. 
For the second backup the file scan time is normal again but after the "finalizing backup" and "deleting unwanted files phase" I get a message regarding a casting error. With this message the backup fails and no restore point is created.

This is the complete error message:

Failed: Cast specificato non valido.
Details: System.InvalidCastException: Cast specificato non valido.
   in System.Data.SQLite.SQLiteDataReader.VerifyType(Int32 i, DbType typ)
   in System.Data.SQLite.SQLiteDataReader.GetInt64(Int32 i)
   in Duplicati.Library.Main.Database.LocalDeleteDatabase.<DropFilesetsFromTable>c__Iterator0.MoveNext()
   in Duplicati.Library.Main.Operation.DeleteHandler.DoRun(LocalDeleteDatabase db, IDbTransaction transaction, Boolean hasVerifiedBacked, Boolean forceCompact)
   in Duplicati.Library.Main.Operation.BackupHandler.Run(String[] sources, IFilter filter)
   in Duplicati.Library.Main.Controller.<Backup>c__AnonStorey0.<>m__0(BackupResults result)
   in Duplicati.Library.Main.Controller.RunAction[T](T result, String[]& paths, Action`1 method)

I already tried to repair database and launch the backup again but I got again the same error.

Someone can help me to solve this issue?

Thanks in Advance,
C0ReDuMP79

**Comment by David-Pa** [comment]
It happened to me too yesterday when I upgraded to 2.0.0.81.
But now it seems to be gone.

**Comment by C0ReDuMP79** [comment]
Hello David-Pa,
I tried to repair and execute backup multiple times but I get always the InvalidCastException error. I tried to downgrade to Duplicati 2.0.0.73 (the previous version I was using), I recovered the old Duplicati's database from the backup, done automatically during the upgrade to the lastest version (2.0.0.81), as kenkendk explained in a post, I ran repair and backup several times but I get always an InvalidCastException, here is the message I get with version 2.0.0.73:

Failed: Impossibile eseguire il cast dell'oggetto da DBNull ad altri tipi.
Details: System.InvalidCastException: Impossibile eseguire il cast dell'oggetto da DBNull ad altri tipi.
   in System.DBNull.System.IConvertible.ToInt64(IFormatProvider provider)
   in Duplicati.Library.Main.Database.LocalDeleteDatabase.<DropFilesetsFromTable>c__Iterator0.MoveNext()
   in Duplicati.Library.Main.Operation.DeleteHandler.DoRun(LocalDeleteDatabase db, IDbTransaction transaction, Boolean hasVerifiedBacked, Boolean forceCompact)
   in Duplicati.Library.Main.Operation.BackupHandler.Run(String[] sources, IFilter filter)
   in Duplicati.Library.Main.Controller.<Backup>c__AnonStorey0.<>m__0(BackupResults result)
   in Duplicati.Library.Main.Controller.RunAction[T](T result, String[]& paths, Action`1 method)

I don't know what can I do to solve this issue, I can't make another full backup, the size is 210 GB over a very slow WAN.

I hope someone can help me to solve this issue so I can continue with incremental backups.

Thanks in Advance,
C0ReDuMP79
